pkg/controller/daemon/daemoncontroller.go

```diff
@@ -899,11 +924,16 @@ func (dsc *DaemonSetsController) nodeShouldRunDaemonPod(node *v1.Node, ds *exten
 		return false, false, false, err
 	}
 
+	// Check if need to fire InsufficientResourceError
+	fireInsufficientResourceError := shouldFireInsufficientResourceError(reasons)
+
 	for _, r := range reasons {
 		glog.V(4).Infof("DaemonSet Predicates failed on node %s for ds '%s/%s' for reason: %v", node.Name, ds.ObjectMeta.Namespace, ds.ObjectMeta.Name, r.GetReason())
 		switch reason := r.(type) {
 		case *predicates.InsufficientResourceError:
-			dsc.eventRecorder.Eventf(ds, v1.EventTypeWarning, FailedPlacementReason, "failed to place pod on %q: %s", node.ObjectMeta.Name, reason.Error())
+			if fireInsufficientResourceError {
+				dsc.eventRecorder.Eventf(ds, v1.EventTypeWarning, FailedPlacementReason, "failed to place pod on %q: %s", node.ObjectMeta.Name, reason.Error())
+			}
 			shouldSchedule = false
```

Comment:
Would we want to return error, or emit event when the reason is intentional, such as `ErrNodeSelectorNotMatch`? If not, we check if it's intentional first. If so, return; else do the other check. Something like:

```go 
for _, r := range reasons {
  // only check intentional predicates, if we find any, return false, false, false, nil
}

for _, r := range reasons { 
  switch r.(type) { 
  case *predicates.InsufficientResourceError:
        // emit event, we know we don't have any intentional predicates at this point 
  case *predicates.PredicateFailureError:
       switch reason {
       // no need to check intentional ones 
       case xxx: // unintentional ones 
       case xxx: // unexpected ones 
       default: 
       }
       if emitEvent {xxx}
  }
}
return 
```